python/lsst/daf/butler/script/queryCollections.py

```diff
@@ -72,37 +97,78 @@ def _getTable(
         names=("Name", typeCol, descriptionCol),
         dtype=(str, str, str),
     )
+    if show_dataset_types:
+        table.add_column(Column(name="Dataset Types", dtype=str))
     butler = Butler.from_config(repo)
+
+    def addDatasetTypes(collection_table: Table, collection: str, dataset_types: list[str]) -> Table:
+        if dataset_types[0] == "":
+            cinfo = butler.collections.get_info(collection, include_summary=True)
+            dataset_types = _parseDatasetTypes(cinfo.dataset_types)
+        if exclude_dataset_types:
+            dataset_types = [
+                dt
+                for dt in dataset_types
+                if not any(fnmatch(dt, pattern) for pattern in exclude_dataset_types)
```

Comment:
It occurs to me that this is quite similar to code I have in pipe_base called `lsst.pipe.base.utils.filter_by_dataset_type_glob` that I should probably move here to cli utils and have an option to negate.